### ISSUE TITLE
[Bug] pysearpc.common.SearpcError: invalid path

### ISSUE THREAD

**Issue by weimens** [haiwen/seahub#4752]
When i try to open an shared folder, where i have permission, in an library where i don't have permission i get this exception:
```
[ERROR] django.request:135 handle_uncaught_exception Internal Server Error: /library/091046dc-dd20-4164-b19a-be58dea015d8/invalid_path_bug/test111
Traceback (most recent call last):
  File "/opt/seafile/seafile-pro-server-7.1.8/seahub/thirdpart/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/opt/seafile/seafile-pro-server-7.1.8/seahub/thirdpart/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/opt/seafile/seafile-pro-server-7.1.8/seahub/thirdpart/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/opt/seafile/seafile-pro-server-7.1.8/seahub/thirdpart/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/seafile/seafile-pro-server-7.1.8/seahub/seahub/auth/decorators.py", line 27, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/opt/seafile/seafile-pro-server-7.1.8/seahub/seahub/views/__init__.py", line 1150, in react_fake_view
    if not seafile_api.get_dirent_by_path(converted_repo_id, converted_path):
  File "/opt/seafile/seafile-pro-server-7.1.8/seafile/lib64/python3.6/site-packages/seaserv/api.py", line 236, in get_dirent_by_path
    return seafserv_threaded_rpc.get_dirent_by_path(repo_id, path)
  File "/opt/seafile/seafile-pro-server-7.1.8/seafile/lib64/python3.6/site-packages/pysearpc/client.py", line 128, in newfunc
    return fret(ret_str)
  File "/opt/seafile/seafile-pro-server-7.1.8/seafile/lib64/python3.6/site-packages/pysearpc/client.py", line 66, in _fret_obj
    raise SearpcError(dicts['err_msg'])
pysearpc.common.SearpcError: invalid path
```
Version: seafile-pro-server_7.1.8_x86-64_CentOS

How to reproduce: 
* create an library with user1
* add an folder to that library
* give user2 read/write permission to that folder
* copy the internal smart link to the folder
* open the smart link with user2

Current behavior:
* Django Error Page

Expected behavior:
* Content of folder

**Comment by weimens** [comment]
Some additional information. I can trace the error back to https://github.com/haiwen/seafile-server/blob/11b3fc45e3da9e074d3c5407f3b531356b01c8fa/common/rpc-service.c#L3236 looks like there is an intentional check for the case that get_dirent_by_path gets called with "/" as the path, though that is exactly what happens when I try to share an internal smart link to an folder. 
If I disable the seafile_api.get_dirent_by_path check in seahub/views/__init__.py:react_fake_view everything works fine. Maybe it helps to narrow down the problem. I don’t have not enough insight in seafile to understand way the check for disallowing "/"  as the path is here in the first place.

**Comment by imwhatiam** [comment]
Hello.

Thanks for reporting this bug, we will fix it in the next 9.0.x version.

**Comment by weimens** [comment]
Thank you, I'm still on seafile 8. But the fix https://github.com/haiwen/seahub/pull/5161 works for me.
